test/core/miner/commit_batcher_test.cpp

```diff
@@ -0,0 +1,258 @@
+  using vm::message::SignedMessage;
+  using vm::message::UnsignedMessage;
+  using PairStorage = CommitBatcherImpl::PairStorage;
+  using MapPairStorage = std::map<SectorNumber, PairStorage>;
+  using fc::mining::types::FeeConfig;
+  using fc::mining::types::PaddedPieceSize;
+  using fc::mining::types::Piece;
+  using fc::mining::types::PieceInfo;
+  using libp2p::basic::ManualSchedulerBackend;
+  using libp2p::basic::Scheduler;
+  using libp2p::basic::SchedulerImpl;
+  using mining::SelectAddress;
+  using proofs::ProofEngineMock;
+  using vm::actor::builtin::types::miner::SectorPreCommitOnChainInfo;
+  using BatcherCallbackMock =
+      std::function<void(const outcome::result<CID> &cid)>;
+
+  MATCHER_P(methodMatcher, method, "compare msg method name") {
+    return (arg.method == method);
+  }
+
+  class CommitBatcherTest : public testing::Test {
+   protected:
+    void SetUp() override {
+      scheduler_backend_ = std::make_shared<ManualSchedulerBackend>();
+      scheduler_ = std::make_shared<SchedulerImpl>(scheduler_backend_,
+                                                   Scheduler::Config{});
+
+      miner_id_ = 1;
+      miner_address_ = Address::makeFromId(miner_id_);
+      side_address_ = Address::makeFromId(++miner_id_);
+
+      api::BlockHeader block;
+      block.height = 2;
+
+      tipset_ = std::make_shared<Tipset>(
+          TipsetKey(), std::vector<api::BlockHeader>({block}));
+
+      EXPECT_CALL(mock_ChainHead, Call())
+          .WillRepeatedly(testing::Return(tipset_));
+
+      fee_config_ = std::make_shared<FeeConfig>();
+      fee_config_->max_precommit_batch_gas_fee.base =
+          TokenAmount{"50000000000000000"};
+      fee_config_->max_precommit_batch_gas_fee.per_sector =
+          TokenAmount{"250000000000000"};
+
+      EXPECT_CALL(mock_StateMinerInitialPledgeCollateral,
+                  Call(miner_address_, _, _))
+          .WillRepeatedly(testing::Return(TokenAmount(100)));
+
+      EXPECT_CALL(mock_StateSectorPreCommitInfo, Call(miner_address_, _, _))
+          .WillRepeatedly(testing::Return(SectorPreCommitOnChainInfo()));
+      proof_ = std::make_shared<ProofEngineMock>();
+
+      EXPECT_CALL(mock_StateMinerInfo, Call(_, _))
+          .WillRepeatedly(testing::Return(MinerInfo()));
+
+      EXPECT_CALL(mock_ChainGetTipSet, Call(_))
+          .WillRepeatedly(testing::Return(tipset_));
+
+      callback_mock_ = [](const outcome::result<CID> &cid) -> void {
+        EXPECT_TRUE(cid.has_value());
+      };
+    }
+
+    std::shared_ptr<FullNodeApi> api_ = std::make_shared<FullNodeApi>();
+    std::shared_ptr<Scheduler> scheduler_;
+    MapPairStorage pair_storage_;
+    std::shared_ptr<Tipset> tipset_;
+    Address miner_address_;
+    Address side_address_;
+    Address wrong_side_address_;
+    ActorId miner_id_;
+    std::shared_ptr<ProofEngineMock> proof_;
+    std::shared_ptr<FeeConfig> fee_config_;
+    BatcherCallbackMock callback_mock_;
```

Comment:
```suggestion
    MockStdFunction<CommitCallback> callback_mock_;
```
[example](https://github.com/filecoin-project/cpp-filecoin/blob/master/test/core/markets/storage/chain_events/chain_events_test.cpp#L143-L145)